service/src/test/java/org/sunbird/metrics/actors/CourseMetricsActorTest.java

```diff
@@ -1,93 +1,117 @@
+@PowerMockIgnore("javax.management.*")
 public class CourseMetricsActorTest {
 
   private static ActorSystem system;
   private static final Props props = Props.create(CourseMetricsActor.class);
   private static String userId = "dnk298voopir80249";
   private static String batchId = "jkwf6t3r083fp4h";
   private static final String orgId = "vdckcyigc68569";
+  private static Map<String, Object> infoMap = new HashMap<>();
+  private static Map<String, Object> userOrgMap = new HashMap<>();
+  private static final String HTTP_POST = "POST";
+  private static ObjectMapper mapper = new ObjectMapper();
+  private static CassandraOperationImpl cassandraOperation;
 
   @BeforeClass
   public static void setUp() {
-    SunbirdMWService.init();
     system = ActorSystem.create("system");
-    Util.checkCassandraDbConnections(JsonKey.SUNBIRD);
-    insertBatchDataToES();
-    insertOrgDataToES();
-    insertUserCoursesDataToES();
-    insertUserDataToES();
-  }
-
-  private static void insertUserDataToES() {
-    Map<String, Object> userMap = new HashMap<>();
-    userMap.put(JsonKey.USER_ID, userId);
-    userMap.put(JsonKey.FIRST_NAME, "alpha");
-    userMap.put(JsonKey.ID, userId);
-    userMap.put(JsonKey.ROOT_ORG_ID, "ORG_001");
-    userMap.put(JsonKey.USERNAME, "alpha-beta");
-    userMap.put(JsonKey.ROOT_ORG_ID, orgId);
-    userMap.put(JsonKey.EMAIL, "arvind.glaitm108@gmail.com");
-    ElasticSearchUtil.createData(
-        EsIndex.sunbird.getIndexName(), EsType.user.getTypeName(), userId, userMap);
-  }
-
-  private static void insertBatchDataToES() {
-    Map<String, Object> batchMap = new HashMap<>();
-    batchMap.put(JsonKey.ID, batchId);
-    ElasticSearchUtil.createData(
-        EsIndex.sunbird.getIndexName(), EsType.course.getTypeName(), batchId, batchMap);
-  }
-
-  private static void insertUserCoursesDataToES() {
-    Map<String, Object> userCoursesMap = new HashMap<>();
-    userCoursesMap.put(JsonKey.ID, batchId + JsonKey.PRIMARY_KEY_DELIMETER + userId);
-    userCoursesMap.put(JsonKey.BATCH_ID, batchId);
-    userCoursesMap.put(JsonKey.USER_ID, userId);
-    userCoursesMap.put(JsonKey.ROOT_ORG_ID, orgId);
-    userCoursesMap.put(JsonKey.PROGRESS, 1);
-    ElasticSearchUtil.createData(
-        EsIndex.sunbird.getIndexName(),
-        EsType.usercourses.getTypeName(),
-        batchId + JsonKey.PRIMARY_KEY_DELIMETER + userId,
-        userCoursesMap);
+    infoMap.put(JsonKey.FIRST_NAME, "user_first_name");
+    infoMap.put(JsonKey.BATCH_ID, "batch_123");
   }
 
-  private static void insertOrgDataToES() {
-    Map<String, Object> orgMap = new HashMap<>();
-    orgMap.put(JsonKey.ID, orgId);
-    orgMap.put(JsonKey.ORGANISATION_NAME, "IIM");
-    orgMap.put(JsonKey.ORGANISATION_ID, orgId);
-    ElasticSearchUtil.createData(
-        EsIndex.sunbird.getIndexName(), EsType.organisation.getTypeName(), batchId, orgMap);
+  @Before
+  public void before() {
+    Map<String, Object> esComplexSearchMap = new HashMap<>();
```

Comment:
Break this long method into two functions.

1) mockESComplexSearch
2) mockESGetDataByIdentifier